assets/queries/k8s/statefulset_without_pod_disruption_budget/query.rego

```diff
@@ -3,26 +3,28 @@ package Cx
 CxPolicy[result] {
 	statefulset := input.document[i]
 	statefulset.kind == "StatefulSet"
+	statefulset.spec.replicas > 1
 	metadata := statefulset.metadata
 
-	not CheckIFPdbExists(statefulset)
+	CheckIFPdbExists(statefulset) == false
 
 	result := {
 		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name=%s", [metadata.name]),
+		"searchKey": sprintf("metadata.name={{%s}}.spec.selector.matchLabels", [metadata.name]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("metadata.name=%s is not targeted by a PDB", [metadata.name]),
-		"keyActualValue": sprintf("metadata.name=%s is not targeted by a PDB", [metadata.name]),
+		"keyExpectedValue": sprintf("metadata.name=%s is targeted by a PodDisruptionBudget", [metadata.name]),
+		"keyActualValue": sprintf("metadata.name=%s is not targeted by a PodDisruptionBudget", [metadata.name]),
 	}
 }
 
-CheckIFPdbExists(statefulsets) = result {
+CheckIFPdbExists(statefulset) = result {
 	documents := input.document
 	pdbs := [pdb | documents[index].kind == "PodDisruptionBudget"; pdb = documents[index]]
```

Comment:
Don't iterate through every document, the query should look for PodDisruptionBudget inside the same document to prevent timeouts